src/Initializer/InputParameters.hpp

```diff
@@ -0,0 +1,196 @@
+
+#ifndef INPUT_PARAMETERS_HPP_
+#define INPUT_PARAMETERS_HPP_
+
+#include <cstdint>
+#include <string>
+#include <array>
+#include <yaml-cpp/yaml.h>
+
+#include <xdmfwriter/XdmfWriter.h>
+
+#include "Geometry/MeshReader.h"
+#include "SourceTerm/Manager.h"
+#include "Checkpoint/Backend.h"
+#include "time_stepping/LtsWeights/WeightsFactory.h"
+
+namespace seissol::initializer::parameters {
+// constexpr auto NUMBER_OF_QUANTITIES = tensor::Q::Shape[ sizeof(tensor::Q::Shape) /
+// sizeof(tensor::Q::Shape[0]) - 1];
+
+constexpr bool modelAnelastic() { return NUMBER_OF_RELAXATION_MECHANISMS > 0; }
+
+constexpr bool modelPoroelastic() {
+#ifdef USE_POROELASTIC
+  return true;
+#else
+  return false;
+#endif
+}
+
+constexpr bool modelAnisotropic() {
+#ifdef USE_ANISOTROPIC
+  return true;
+#else
+  return false;
+#endif
+}
+
+struct ModelParameters {
+  double gravitationalAcceleration = 9.81;
+  double tv = 0.1;
+  bool plasticity = false;
+  bool useCellHomogenizedMaterial = false;
+  double freqCentral = 0.0;
+  double freqRatio = 0.0;
+  std::string materialFileName = "";
+  std::string boundaryFileName = "";
+};
+
+enum class InitializationType {
+  Zero,
+  Planarwave,
+  SuperimposedPlanarwave,
+  Travelling,
+  Scholte,
+  Snell,
+  Ocean0,
+  Ocean1,
+  Ocean2
+};
+
+struct InitializationParameters {
+  InitializationType type;
+  std::array<double, 3> origin;
+  std::array<double, 3> kVec;
+  std::array<double, NUMBER_OF_QUANTITIES> ampField;
+};
+
+struct DynamicRuptureParameters {
+  bool hasFault;
+  // TODO: port rest of the DR parameters here?
+};
+
+enum class OutputFormat { None = 10, Xdmf = 6 };
```

Comment:
There is now `WavefieldOutput` as a field (and also a `Checkpoint` field, working as a guard for checkpointing). Will be added to the docu.

We could retire `OutputFormat`—however, this would most likely constitute a breaking change. I've put in a notice that it'll still work, but it is better to use `WavefieldOutput`, and the `outputformat` parameter may be removed in the future (most likely when we introduce a new format anyways).